test/metabase/db/internal_user_test.clj

```diff
@@ -0,0 +1,41 @@
+(ns ^:mb/once metabase.db.internal-user-test
+  (:require
+   [clojure.string :as str]
+   [clojure.test :refer :all]
+   [metabase.config :as config]
+   [metabase.test :as mt]
+   [metabase.util :as u]
+   [toucan2.core :as t2]))
+
+(deftest internal-user-is-in-all-users-test
+  (testing "The internal user should be in the All Users group, just like any other user."
+    (let [memberships  (-> (t2/select-one :model/User config/internal-mb-user-id)
+                           (t2/hydrate :user_group_memberships)
+                           :user_group_memberships
+                           (->> (map :id)))
+          all-users-id (t2/select-one-pk :model/PermissionsGroup :name "All Users")]
+      (is [all-users-id] memberships))))
+
+(deftest internal-user-is-unmodifiable-via-api-test
```

Comment:
This was moved from enterprise/backend/test/metabase_enterprise/internal_user_test.clj